enterprise/models.py

```diff
@@ -321,6 +368,147 @@ def __repr__(self):
         return self.__str__()
 
 
+class UserDataSharingConsentAudit(TimeStampedModel):
+    """
+    Object that exists to store the canonical state of whether a particular
+    user has given consent for their course data to be shared with a particular
+    single-sign-on identity provider
+    """
+    class Meta(object):
+        app_label = 'enterprise'
+        verbose_name = "Data Sharing Consent Audit State"
+        verbose_name_plural = "Data Sharing Consent Audit States"
+        unique_together = ('user', 'enterprise_customer',)
+
+    NOT_SET = 'not_set'
+    ENABLED = 'enabled'
+    DISABLED = 'disabled'
+    STATE_CHOICES = (
+        (NOT_SET, 'Not set'),
+        (ENABLED, 'Enabled'),
+        (DISABLED, 'Disabled'),
+    )
+
+    user = models.ForeignKey(
+        User,
+        on_delete=models.CASCADE,
+    )
+
+    enterprise_customer = models.ForeignKey(
+        EnterpriseCustomer,
+        on_delete=models.CASCADE,
+    )
+
+    state = models.CharField(
+        max_length=8,
+        blank=False,
+        choices=STATE_CHOICES,
+        default=NOT_SET,
+        help_text=_(
+            "Stores whether the user linked to this model has consented to have "
+            "their information shared with the linked EnterpriseCustomer."
+        )
+    )
```

Comment:
@mattdrayer Are we planning to add a `course_id` field to this consent audit model eventually, so that users can consent on a per-course basis (at enrollment), as well as on the current per-enterprise-for-all-courses basis? I don't see a story for that in the backlog, but I seem to remember talking about that.